pkg/k8s/graceful_shutdown.go

```diff
@@ -0,0 +1,74 @@
+package k8s
+
+import (
+	"context"
+	"time"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+)
+
+var (
+	stopCH = make(chan struct{}, 0)
+	log    = logf.Log
+)
+
+// StartStopSignalHandler starts gorutine which is waiting for
+// termination signal and returns chan for indication when operator
+// can really stop.
+func StartStopSignalHandler(client client.Client, namespaces []string) <-chan struct{} {
+	go handleStopSignal(client, namespaces)
+	return stopCH
+}
+
+func handleStopSignal(client client.Client, namespaces []string) {
+	<-signals.SetupSignalHandler()
+	stop(client, namespaces)
+	close(stopCH)
+}
+
+// Stop is used to understand, when we need to stop operator(usially SIGTERM)
+// to start cleanup process and delete required pxc clusters in current(operator)
+// namespace. See K8SPXC-529
+func stop(cl client.Client, namespaces []string) {
+	log.Info("Got stop signal, starting to list clusters")
+
+	for {
```

Comment:
Should we have some timeout here? Not sure but I assume that loop may be infinite.

Reply:
pod will be force killed after timeout, which is configurable from yaml files